lib/vagrant/action/builtin/cloud_init_wait.rb

```diff
@@ -0,0 +1,26 @@
+module Vagrant
+  module Action
+    module Builtin
+      class CloudInitWait
+
+        def initialize(app, env)
+          @app    = app
+          @logger = Log4r::Logger.new("vagrant::action::builtin::cloudinitwait")
+        end
+
+        def call(env)
+          machine = env[:machine]
+          if !machine.config.vm.cloud_init_configs.empty?
+            if machine.communicate.test("cloud-init -v", error_check: false)
```

Comment:
can use this instead to check for the command and not rely on exit code to be zero:

```ruby
if machine.communicate.test("command -v cloud-init")
```